lib/rspec/retry.rb

```diff
@@ -48,6 +50,10 @@ def self.apply
               break unless exceptions_to_retry.include?(example.exception.class)
             end
 
+            if exceptions_to_ignore_retry.any?
```

Comment:
no need for this wrapping if statement (it's needed above because the default for `exceptions_to_retry` is all exceptions)